internal/storage/storage.go

```diff
@@ -0,0 +1,213 @@
+package storage
+
+import (
+	"context"
+	"database/sql"
+	"errors"
+	"os"
+	"time"
+
+	// sqlite3 driver
+	_ "github.com/mattn/go-sqlite3"
+)
+
+const (
+	querySaveSnapshot = `INSERT INTO snapshots(cluster_name, created_at, data) 
+							VALUES(?, ?, ?)`
+	querySaveRecoveries = `INSERT INTO recoveries(cluster_name, created_at, data) 
+							VALUES(?, ?, ?)`
+	queryCreateTableSnapshots = `CREATE TABLE IF NOT EXISTS snapshots (
+		"id" integer NOT NULL PRIMARY KEY AUTOINCREMENT,		
+		"cluster_name" TEXT,
+		"created_at" INTEGER,
+		"data" BLOB
+	  )`
+	queryCreateTableRecoveries = `CREATE TABLE IF NOT EXISTS recoveries (
+		"id" integer NOT NULL PRIMARY KEY AUTOINCREMENT,		
+		"cluster_name" TEXT,
+		"created_at" INTEGER,
+		"data" BLOB
+	  )`
+	queryGetLastSnapshot = `SELECT data
+		FROM snapshots
+		WHERE cluster_name = ?
+		ORDER BY id DESC limit 1`
+	queryGetRecoveries = `SELECT data
+		FROM recoveries
+		WHERE cluster_name = ?`
+)
+
+var (
+	ErrEmptyResult = errors.New("empty result")
+)
+
+type SaveRequest struct {
+	ClusterName string
+	CreatedAt   int64
+	Data        []byte
+}
+
+type Storage interface {
```

Comment:
You have an interface of your repository. Move the concrete implementation to another file `sqlite.go` and add some constructors

```go
type Storage interface { ... }

func NewSQLiteStorage(cfg Config) (*SQLiteStorage, error) { ... }

func NewMockStorage() (*MockStorage, error) { ... }
```

You might move implementations to its own packages.